gwf/cli/arg_parsing.py

```diff
@@ -0,0 +1,55 @@
+from __future__ import absolute_import, print_function
+from ..utils import cache, import_object
+from ..core import PreparedWorkflow
+import argparse
+
+
+class SubCommand(object):
+    """Super-class for sub-commands."""
+
+    # Class variable used to get the active workflow.
+    _workflow = None
+    @property
+    @cache
+    def workflow(cls):
+        if cls._workflow is None: # pragma: no cover
+            sys.exit("Workflow not specified") # FIXME: exception, but it really should never happen since we have a default
+        return PreparedWorkflow(import_object(cls._workflow))
+
+    def set_arguments(self, subparser):
+        "Provide the parser options for the sub-command."
+
+    def handle(self, arguments):
+        "Callback when this sub-command is invoked."
+
+class ArgumentDispatching(object):
+    """Class responsible for setting up the command line arguments and dispatching
+    to sub-commands when invoked."""
+
+    def __init__(self):
+        self.parser = argparse.ArgumentParser(
+            description="Grid WorkFlow",
+        )
+
+        # Set global options here. Options for sub-commands will be set by the
+        # sub-commands we dispatch to.
+        self.parser.add_argument("-w", "--workflow",
+                                 help = "workflow file/object (default 'workflow.py')",
+                                 default = "workflow.py")
```

Comment:
`default='workflow.py:gwf'` makes it clear how to specify another workflow object.